src/txdb.cpp

```diff
@@ -115,8 +115,7 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
             changed++;
         }
         count++;
-        CCoinsMap::iterator itOld = it++;
-        mapCoins.erase(itOld);
+        it = erase ? mapCoins.erase(it) : ++it;
```

Comment:
Should `++it` be outside the ternary conditional here?

Reply:
No. `std::unordered_map::erase()` returns an iterator to the next item in the map: https://en.cppreference.com/w/cpp/container/unordered_map/erase.

Reply:
Oh -- and it appears to be a convention. Thanks @jnewbery.

Reply:
`it = ++it;` is kind of strange though. How about `it = mapCoins.erase(it) : std::next(it);` ?

Reply:
Will change if I rebase.

Reply:
Fixed, thanks.